### ISSUE TITLE
Suggestion (rule change): Add optional chaining support to no-unneeded-ternary

### ISSUE THREAD

**Issue by BlueRaja** [eslint/eslint#13819]
**What rule do you want to change?**
no-unneeded-ternary

**Does this change cause the rule to produce more or fewer warnings?**
more

**How will the change be implemented? (New option, new default behavior, etc.)?**
new default behavior

**Please provide some example code that this change will affect:**

```js
// bad
const a = b ? b.c : null;

// good
const a = b?.c;
```

**What does the rule currently do for this code?**
Nothing

**What will the rule do after it's changed?**
(see above)

**Comment by mdjermanovic** [comment]
Hi @BlueRaja, thanks for the issue!

I think this belongs to [prefer-optional-chaining](https://github.com/eslint/eslint/issues/13430) rule, but we should discuss some details, since `b?.c` is equivalent to `b != null ? b.c : undefined`. Depending on `b`, in might not be same as `b ? b.c : null`.

**Comment by nzakas** [comment]
I also think this rule should stay focused on the ternary operator.
